components/tidb_query/src/util.rs

```diff
@@ -28,9 +21,14 @@ impl MySQLRng {
 
     pub fn new_with_seed(seed: i64) -> Self {
         let temp = seed as u32;
-        let seed1 = (i64::from(temp)).wrapping_mul(0x10001).wrapping_add(55555555) as u32;
+        let seed1 = (i64::from(temp))
```

Comment:
Why do you cast it back to i64? Is there any concern?

Reply:
Because if we don't use i64 it will over flow.

Reply:
But since wrapping is indicated, I think it should be ok? Because from my understanding of the MySQL source code, they manipulate the seed by u32 all the way through.

Reply:
But it's fine once we pass the tests.

Reply:
I refactor code removed useless convert.